choria/framework.go

```diff
@@ -269,25 +268,37 @@ func (fw *Framework) FederationMiddlewareServers() (servers srvcache.Servers, er
 
 // PuppetDBServers resolves the PuppetDB server based on configuration of _x-puppet-db._tcp
 func (fw *Framework) PuppetDBServers() (servers srvcache.Servers, err error) {
-	configured := ""
-	if fw.Config.Choria.PuppetDBHost != "" {
-		configured = fw.Config.Choria.PuppetDBHost
-		if fw.Config.Choria.PuppetDBPort != 0 {
-			configured += ":" + strconv.Itoa(fw.Config.Choria.PuppetDBPort)
-		}
+	defaultHost := fw.Config.Choria.PuppetDBHost
```

Comment:
So maybe we do a few things here:

 * We remove the defaults from the config, so puppetdb settings are empty 
 * We change this method to:

``
if puppetdb_set
   return configured

if srv_enabled
   attempt to resolve via SRV
   return if found

return "puppet:1234"
```

I am happy with hard coding these defaults here, this makes it so srv is only done when not set and trusts configured values.

what do you think?

Reply:
:+1: I updated the PR to match this behavior.